Draft-Accepted/RFCXXXX-FeedbackProviders.md

```diff
@@ -0,0 +1,192 @@
+
+```powershell
+PS> rm -rf
+Remove-Item: A parameter cannot be found that matches parameter name 'rf'.
+[Compatibility]
+  It appears you are trying to recursively and forcibly remove files. In PowerShell, the correct syntax would be:
+    ➤ rm -r -fo
+```
+
+It is up to the feedback provider to decide what error specifically it tries to trigger on.
+Currently the entire error record object gets passed to feedback providers so all the information
+available in that object is available for the feedback providers to trigger on.
+
+____
+
+## Interacting with Feedback Providers
+
+### Customizability
+
+Currently if a user wants to enable/disable particular feedback providers they need to install the
+module and then have it imported in their session. Order of import also determines the ordering when
+multiple feedback providers trigger on the same event, **except** for the `general` feedback
+provider that triggers on `CommandNotFound` exception. This provider will be on by default and be
+the first in the list of suggestions after the execption occurs. For ordering and enabling by
+default we encourage the experience of keeping feedback provider modules imported in the users
+`$PROFILE`. I.e
+```powershell
+# In $PROFILE
+Import-Module FeedbackProvider1
+Import-Module FeedbackProvider2
+```
+Then in the CLI:
+```powershell
+PS>something to trigger feedback providers
+[FeedbackProvider1]
+  Header 1:
+    ➤ Action
+[FeedbackProvider2]
+  Header2:
+    ➤ Action
+```
+
+
+### Predictor Tie in
+
+As a way to enhance the CLI users workflow, we want to encourage feedback providers to utilize the `ICommandPredictor` interface as well. This way the workflow can be feedback provider triggered _then_ user immediately gets recommended command predicted to them. Below is an example taken from the [`command-not-found`](https://github.com/PowerShell/command-not-found) feedback provider.
+
+
+![Alt](../assets/FeedbackProviders/command-not-found.gif)
+
+
+## Goals/Non-Goals
+
+Goals
+- Create a new, consistent, UI for assistive feedback for errors, comments, successful executions, and CommandNotFound exceptions.
+- Expose necessary information releveant to giving assistence to the user via an interface
+- Enable user customizablity of feedback providers to display and order
+- Make creation of feedback providers easy for module developers
+- Enable the creation of an ecosystem of assistive providers
+
+Non-Goals
+- Be a disruptive experience in PowerShell
+- Ship many feedback providers by default 
+
+## Open considerations
+Questions
+- Should the PowerShell engine be responsible for restricting feedback providers from being disruptive?
+
+    - I.e if a feedback provider triggers on **success** _every_ time a user types a particular command, then should the engine restrict that if its never subsequently typed after being provided?
```

Comment:
I think we need some mechanism for the user to suppress being informed multiple times for the same thing.  It probably doesn't make sense for each FeedbackProvider to come up with their own mechanism.